packages/manager/modules/pci/src/projects/project/edit/edit.controller.js

```diff
@@ -0,0 +1,99 @@
+import pick from 'lodash/pick';
+
+import { DEFAULT_PROJECT_KEY, MESSAGES_CONTAINER_NAME } from './edit.constant';
+
+export default class ProjectEditController {
+  /* @ngInject */
+  constructor($stateParams, $translate, CucCloudMessage, ovhUserPref, OvhApiCloudProject) {
+    this.$stateParams = $stateParams;
+    this.$translate = $translate;
+    this.CucCloudMessage = CucCloudMessage;
+    this.ovhUserPref = ovhUserPref;
+    this.OvhApiCloudProject = OvhApiCloudProject;
+
+    this.loading = {
+      init: false,
+      submit: false,
+    };
+
+    this.$ngInit();
+  }
+
+  $ngInit() {
+    this.loading.init = true;
+    this.messageHandler = this.CucCloudMessage.subscribe(MESSAGES_CONTAINER_NAME, {
+      onMessage: () => this.refreshMessage(),
+    });
+
+    return this.OvhApiCloudProject
+      .v6()
+      .get({
+        serviceName: this.$stateParams.projectId,
+      })
+      .$promise
+      .then((project) => {
+        this.project = project;
+      })
+      .then(() => this.ovhUserPref.getValue(DEFAULT_PROJECT_KEY))
+      .then((defaultProject) => {
+        this.defaultProject = defaultProject;
+        this.isDefault = this.$stateParams.projectId === defaultProject.projectId;
+      })
+      .catch((err) => {
+        if (err.status === 404) {
+          return null;
+        }
+        throw err;
+      })
+      .finally(() => {
+        this.loading.init = false;
+      });
+  }
+
+  refreshMessage() {
+    this.messages = this.messageHandler.getMessages();
+  }
+
+  submit() {
+    this.loading.submit = true;
+
+    return this.OvhApiCloudProject
+      .v6()
+      .put(
+        {
+          serviceName: this.$stateParams.projectId,
```

Comment:
This should probably be stored in as an attribute (if ever we want to change the way of fetching this information it will be easier)